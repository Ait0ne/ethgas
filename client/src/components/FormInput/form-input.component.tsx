import React, {InputHTMLAttributes} from 'react';
import {Container} from './form-input.styles';


interface InputProps {
    handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
    label?: string;
    clear?: () => void
}



const FormInput: React.FC<InputProps&InputHTMLAttributes<HTMLInputElement>> = ({handleChange, label, clear, ...otherProps}) => {
    return (
    <Container>
        <input onChange={handleChange} {...otherProps}/>
        {clear&&otherProps.value?.toString().length&&otherProps.value?.toString().length>0?
            <span onClick={clear}>&#x2716;</span>
            :null
            }
        {label ?
        (<label
        className={`${otherProps.value?.toString()&&otherProps.value?.toString().length>0 ? 'shrink': ''}`}
        >
            {label}
        </label> )
        : null
        }
    </Container>
    )
}

export default FormInput;