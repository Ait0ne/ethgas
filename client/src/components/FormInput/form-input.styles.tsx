import styled from 'styled-components';

export const Container = styled.div`
position: relative;
margin: 20px 0px 15px 0px;
width:253px;

>input {
    background: none;
    background-color: white;
    color: grey;
    font-size: 18px;
    padding: 10px 30px 10px 5px;
    display: block;
    border: none;
    border-radius: 4px;
    width:253px;
    position: relative;
    &:focus ~ label {
        top: -26px;
        font-size: 18px;
        color: #fafafa;
    }    
}
>label {
    color: grey;
    font-size: 16px;
    top: 10px;
    font-weight: normal;
    position: absolute;
    pointer-events: none;
    left: 5px;
    transition: 300ms ease all;
    &.shrink {
        top: -26px;
        font-size: 18px;
        color: #fafafa;
    }
}
>span {
    cursor: pointer;
    position: absolute;
    right: 8px;
    font-size: 1.2rem;
    font-weight: bold;
    top: calc(50% - 0.6rem - 5px)

}
`