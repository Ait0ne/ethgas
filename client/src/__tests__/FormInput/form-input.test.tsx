import React from 'react';
import {shallow} from 'enzyme';
import FormInput from '../../components/FormInput/form-input.component';


describe('formInput', () => {
    const onChange = jest.fn()
    const formInput:any = shallow(<FormInput handleChange={onChange} label='email'/>)

    it('renders correctly', () => {
        expect(formInput).toMatchSnapshot()
    } )

    it('calls change function when typed in', () => {
        const input = formInput.find('input')
        input.simulate('change', {target: {value: 'abc'}})

        formInput.update()
        expect(onChange).toHaveBeenCalledWith({target: {value: 'abc'}})
    })

})
