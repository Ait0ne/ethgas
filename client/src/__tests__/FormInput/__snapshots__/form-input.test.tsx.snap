// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formInput renders correctly 1`] = `
<styled.div>
  <input
    onChange={[MockFunction]}
  />
  <label
    className=""
  >
    email
  </label>
</styled.div>
`;
